pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -262,29 +262,39 @@ protected void internalDeleteNamespace(AsyncResponse asyncResponse, boolean auth
 
         // remove from owned namespace map and ephemeral node from ZK
         final List<CompletableFuture<Void>> futures = Lists.newArrayList();
-        try {
-            // remove system topics first.
-            if (!topics.isEmpty()) {
-                for (String topic : topics) {
-                    pulsar().getBrokerService().getTopicIfExists(topic).whenComplete((topicOptional, ex) -> {
-                        topicOptional.ifPresent(systemTopic -> futures.add(systemTopic.deleteForcefully()));
+        // remove system topics first.
+        if (!topics.isEmpty()) {
+            for (String topic : topics) {
+                pulsar().getBrokerService().getTopicIfExists(topic).whenComplete((topicOptional, ex) -> {
```

Comment:
In this case, we have only found the system topic if the system topic is owned by this broker, however, if the system topic is not owned by this broker, we can't find it right?

Reply:
Sorry for the late response.
Yes, I have fixed this by calling adminClient. Thanks for reveiwing.